ostd/src/cpu/id.rs

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! CPU identification numbers.
+
+pub use current::PinCurrentCpu;
+
+/// The ID of a CPU in the system.
+///
+/// If converting from/to an integer, the integer must start from 0 and be less
+/// than the number of CPUs.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct CpuId(u32);
+
+impl CpuId {
+    /// Creates a new instance.
+    ///
+    /// # Panics
+    ///
+    /// The given number must be within the range `[0..ostd::cpu::num_cpus()]`;
+    /// otherwise, the method will panic.
+    pub fn new(raw_id: u32) -> Self {
+        assert!(raw_id < num_cpus() as u32);
+        // SAFETY: the above assertion ensures that the raw number is valid.
+        unsafe { Self::new_unchecked(raw_id) }
+    }
+
+    /// Creates a new instance.
+    ///
+    /// # Safety
+    ///
+    /// The given number must be within the range `[0..ostd::cpu::num_cpus()]`.
```

Comment:
This is incorrect. The number must be smaller than `num_cpus()`, so it cannot equal to `num_cpus()`.
```suggestion
    /// The given number must be smaller than the total number of CPUs ([`ostd::cpu::num_cpus()`]).
```

Update the doc comments of `CpuId::new` as well.